jupyterlab_gcedetails/src/components/widget_popup.tsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import { Popper, Fade, Paper } from '@material-ui/core';
+import { classes } from 'typestyle';
+import { STYLES } from '../data';
+
+interface Props {
+  children: React.ReactNode;
+}
+
+const GRAPH_ICON_CLASS = 'jp-UtilizationGraphsIcon';
+
+export function WidgetPopup(props: Props) {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```

Comment:
In general I haven't used `useState` and stuck with the functional component style. In those cases I mostly just stuck with class-based components but I'm perfectly fine with this.

Reply:
In that case, I'll change this component to be a class-based component to better match the rest of the code then :)